### ISSUE TITLE
Cursor moving when I go to the bottom left of the screen.

### ISSUE THREAD

**Issue by JJJHHHCCC** [ppy/osu#14186]
https://user-images.githubusercontent.com/88674997/128737288-b66ff8b5-bd05-4882-8955-0b4c1696a37d.mp4

**Describe the bug:
The mouse goes to the corner of the screen at 4 seconds into the video and 15 seconds.
Map: https://osu.ppy.sh/beatmapsets/1077064#osu/2253886** 

**osu!lazer version: 2021.809.0-lazer** 

**Logs:
[runtime.log](https://github.com/ppy/osu/files/6955841/runtime.log)
[performance.log](https://github.com/ppy/osu/files/6955843/performance.log)
[updater.log](https://github.com/ppy/osu/files/6955845/updater.log)
[database.log](https://github.com/ppy/osu/files/6955846/database.log)
[network.log](https://github.com/ppy/osu/files/6955847/network.log)
I don't know which one you need.**

**Comment by Theighlin** [comment]
duplicate of #14021

**Comment by smoogipoo** [comment]
This doesn't look like a duplicate? The mentioned issue is for Linux users whereas this is Windows, and this one seems much more severe.

@JJJHHHCCC can you try disabling "high precision mouse" in settings?

**Comment by BriccioCulicio** [comment]
Having the same problem on this osu!lazer version: 2022.528.0-lazer
But my cursor, while in gameplay, changes to the menu cursor.
It suddenly snaps my cursor to the bottom of the screen, mostly left side, though it happened once on the right as well and after that it changes the normal gameplay cursor to the menu one, and I cannot click anything.
I am playing on Windows 10, with keyboard and mouse
Here are two examples:

https://user-images.githubusercontent.com/61429656/171881470-7e106936-3c85-40ad-b389-db7bc29b2ece.mp4

https://user-images.githubusercontent.com/61429656/171881976-7a73d085-aca2-4440-99ff-57f551d841d6.mp4

Here are my logs:
[network.log](https://github.com/ppy/osu/files/8833286/network.log)
[performance.log](https://github.com/ppy/osu/files/8833287/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8833288/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8833289/updater.log)
[database.log](https://github.com/ppy/osu/files/8833290/database.log)

**Comment by peppy** [comment]
@Susko3 interested in your thoughts on this one. Looks like a classic case of SDL warping gone wrong.

**Comment by Susko3** [comment]
1. The gameplay cursor turns into the menu cursor and hit objects don't recognise input when `IWindow.CursorInWindow = false` (checked by manually pinning the value to false).
2. Always on top windows are known to be broken and not play well with SDL/o!f: [ppy/osu-framwork#4351](https://github.com/ppy/osu-framework/issues/4351)
   - although I can't repro that exact issue anymore
3. Looking at the footage frame-by-frame, the OS cursor is visible when the cursor is in the lower ~36 pixels
   - I think this is because the windows taskbar is visible (doesn't render since exclusive fullscreen), always-on-top, and is taking input
4. I can reproduce the exact behaviour shown in the videos by opening an always-on-top Task Manager, positioning it where the taskbar should be, and running osu! in fullscreen.

In short, OS cursor (which isn't strictly tied to the position of the framework cursor) goes over an always on top window, and things break: SDL reports the cursor leaving the window, but when it reenters, `SDL_WINDOWEVENT_ENTER` is not sent. Clicking inside the window sends `SDL_WINDOWEVENT_ENTER` and restores focus and relative mode to expected.

@BriccioCulicio To confirm my hypotheses in 3., could you move your windows taskbar to the top of the screen. In this case, the cursor should break as shown in the videos when near the top of the screen, instead of at the bottom.

As noted in https://github.com/ppy/osu-framework/issues/4351, always on top (AOT) windows are common throughout windows UX. SDL "handles" them by always warping/confining the OS cursor to the center (reasonable handling, as most AOT windows are temporary popups that appear on at the edges) [which doesn't work for us](https://github.com/ppy/osu-framework/pull/5136).

Just for kicks, I tested the windows volume popup (as that is the most common AOT window when playing osu!) and it does in fact break as well. (It will happily take mouse click inputs, even if it's invisible.)

---

This is a weird one. The taskbar should definitely not be active in fullscreen applications (for me, the taskbar can sometimes appear over fullscreen content, like videos in firefox) -- so windows may be to blame here.

We could warp the OS cursor to the center of the window, but this will cause other problems. SDL at one point had the cursor confined within ~75% of the window size when using relative mode (to avoid windows notifications and other AOTs, while not being too aggressive on the warping).

Not really sure what to do here.

**Comment by Susko3** [comment]
> Did you mean having my windows task manager always on top?

If you have an always-on-top Task Manager open things will definitely break, don't do that.

>fixed the issue by having the Automatically hide the taskbar while in desktop mode on

Automatically hiding the taskbar should not be necessary, windows should hide it for you when using fullscreen apps.
 I guess this is a temporary workaround. Weird how it works, the hidden taskbar still takes input (so you have hover to open).

Do you constantly get this issue, or is it only after some time/event (eg. doesn't happen until you alt-tab, etc.)?
Alt-tabbing in and out of osu! might help windows realise that it's actually a fullscreen app, try that and report back!

**Comment by BriccioCulicio** [comment]
This is the second time it happened, first one was like 3 months ago, not sure how I fixed it at that time, I just stopped playing for some time and when I came back it didn't happen anymore.
Alt-tabbing, restarting the game or restarting the computer don't work or at least not this time around . But yeah thanks for the help.
